server/src/main/java/io/spine/server/delivery/FailedReception.java

```diff
@@ -0,0 +1,140 @@
+ * <p>End-users may choose to do one of the following:
+ *
+ * <ul>
+ *     <li>{@linkplain #markDelivered() mark} the message as delivered;
+ *     <li>{@linkplain #repeatDispatching() repeat dispatching} of the message immediately.
+ * </ul>
+ *
+ * <p>Alternatively, end-users may choose to define their own way of reacting
+ * to reception failures by implementing a custom {@code FailedReception.Action},
+ * and returning it via the corresponding
+ * {@linkplain DeliveryMonitor#onReceptionFailure(FailedReception) API} of {@code DeliveryMonitor}.
+ */
+public final class FailedReception {
+
+    private final InboxMessage message;
+    private final Error error;
+    private final Conveyor conveyor;
+    private final RepeatDispatching repeat;
+
+    /**
+     * Creates an instance of the failed reception.
+     *
+     * @param message
+     *         the message which failed to handle
+     * @param error
+     *         the details of the failure
+     * @param conveyor
+     *         the conveyor holding the {@code InboxMessage}s currently being delivered;
+     *         used to manipulate the message upon end-user's choice
+     * @param repeat
+     *         a callback invoked to repeat the dispatching
+     *         of the {@code InboxMessage} immediately
+     */
+    FailedReception(InboxMessage message,
+                    Error error,
+                    Conveyor conveyor,
+                    RepeatDispatching repeat) {
+        this.message = message;
+        this.error = error;
+        this.conveyor = conveyor;
+        this.repeat = repeat;
+    }
+
+    /**
+     * Returns the original {@code InboxMessage}.
+     */
+    public InboxMessage message() {
+        return message;
+    }
+
+    /**
+     * Returns the failure.
+     */
+    public Error error() {
+        return error;
+    }
+
+    /**
+     * Returns an action which marks the message
+     * as {@linkplain InboxMessageStatus#DELIVERED delivered}.
+     *
+     * <p>The message will be automatically removed from its inbox
+     * at the end of the delivery stage.
+     */
+    @SuppressWarnings("WeakerAccess")   /* Part of the public API. */
+    public Action markDelivered() {
+        return () -> conveyor.markDelivered(message);
+    }
+
+    /**
+     * Returns an action immediately repeats the dispatching of the message.
+     */
+    @SuppressWarnings("WeakerAccess")   /* Part of the public API. */
+    public Action repeatDispatching() {
+        return repeat::dispatchAgain;
+    }
+
+    /**
+     * An action to take in relation to the failed reception.
+     */
+    @SPI
+    @FunctionalInterface
+    public interface Action {
+
+        /**
+         * Executes an action.
+         */
+        void execute();
+    }
+
+    /**
+     * An internal interface used to specify the action,
+     * which immediately repeats the {@code InboxMessage} dispatching.
+     */
+    @Internal
+    @FunctionalInterface
+    public interface RepeatDispatching {
```

Comment:
It says that it is an action. Why doesn't it extend the `Action` interface above?